src/extension/module.js

```diff
@@ -251,6 +251,40 @@
     prod: 'host',
   };
 
+/**
+ * log RUM if part of the sample.
+ * @param {string} checkpoint identifies the checkpoint in funnel
+ * @param {Object} data additional data for RUM sample
+ */
+
+// eslint-disable-next-line import/prefer-default-export
+function sampleRUM(checkpoint, collect = false, data = {}) {
+  try {
+    window.hlx = window.hlx || {};
+    if (!window.hlx.rum) {
+      const usp = new URLSearchParams(window.location.search);
+      const weight = 5; 
+      // eslint-disable-next-line no-bitwise
+      const hashCode = (s) => s.split('').reduce((a, b) => (((a << 5) - a) + b.charCodeAt(0)) | 0, 0);
+      const id = `${hashCode(window.location.href)}-${new Date().getDay()}-${new Date().getHours()}-${chrome.runtime.id}`;
+      const random = Math.random();
+      const isSelected = (random * weight <= (collect) ? 5 : 1);
+      // eslint-disable-next-line object-curly-newline
+      window.hlx.rum = { weight, id, random, isSelected };
+    }
+    const { random, weight, id } = window.hlx.rum;
+    if (random && (random * weight <= (collect) ? 5 : 1)) {
+      // eslint-disable-next-line object-curly-newline
+      const body = JSON.stringify({ weight, id, referer: window.location.href, generation: 'sidekick-gen1', checkpoint, src: window.location.pathname, ...data });
```

Comment:
IIUC, the window.location is not the one of the page that is authored, because this is running in the extension worker. @MarquiseRosier did you try this if it works?

Reply:
Also, the `src` attribute isn't processed on the server-side, instead it should be passed in each `sampleRUM` invocations as `source`.